subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/ReportState.kt

```diff
@@ -3,112 +3,151 @@ package com.avito.android.test.report
 import com.avito.android.test.report.model.DataSet
 import com.avito.android.test.report.model.StepResult
 import com.avito.android.test.report.model.TestMetadata
+import com.avito.filestorage.FutureValue
+import com.avito.filestorage.RemoteStorage
 import com.avito.report.model.Entry
 import com.avito.report.model.Incident
 import com.avito.report.model.Video
 
 sealed class ReportState {
 
-    abstract val isFirstStepOrPrecondition: Boolean
-
-    object Nothing : ReportState() {
-        override val isFirstStepOrPrecondition: Boolean = true
-    }
-
-    sealed class Initialized : ReportState() {
-        abstract val testMetadata: TestMetadata
-        abstract var incident: Incident?
-
-        data class Started(
-            override val testMetadata: TestMetadata,
-            override var incident: Incident? = null,
-            var currentStep: StepResult? = null,
-            var stepNumber: Int = 0,
-            var preconditionNumber: Int = 0,
-            var video: Video? = null,
-            var dataSet: DataSet? = null,
-            var startTime: Long,
-            var endTime: Long = 0,
-            var preconditionStepList: MutableList<StepResult> = mutableListOf(),
-            var testCaseStepList: MutableList<StepResult> = mutableListOf()
-        ) : Initialized() {
-
-            override val isFirstStepOrPrecondition: Boolean
-                get() = preconditionNumber == 1 || preconditionNumber == 0 && stepNumber == 1
-
-            internal fun getCurrentStepOrCreate(factory: () -> StepResult): StepResult {
-                return currentStep ?: factory().also { step ->
-                    currentStep = step
-                    when {
-                        testCaseStepList.isNotEmpty() -> testCaseStepList.add(step)
-                        else -> preconditionStepList.add(step)
+    sealed class NotFinished : ReportState() {
+        abstract val entriesBeforeSteps: MutableList<Entry>
+        abstract val uploadsBeforeSteps: MutableList<FutureValue<RemoteStorage.Result>>
+
+        data class NotInitialized(
+            override val entriesBeforeSteps: MutableList<Entry> = mutableListOf(),
+            override val uploadsBeforeSteps: MutableList<FutureValue<RemoteStorage.Result>> = mutableListOf()
+        ) : NotFinished()
+
+        sealed class Initialized : NotFinished() {
+            abstract val testMetadata: TestMetadata
+            abstract var incident: Incident?
+            abstract var incidentScreenshot: FutureValue<RemoteStorage.Result>?
+
+            data class NotStarted(
```

Comment:
I'm thinking about avoiding `Not` in naming states. 
Denial says that it can be anything except one state.

How about naming a transition to the next step?

- `Starting` -> `Started`
- `Finishing` -> `Finished`

It shows explicitly what state will be the next.

Reply:
But we have state `NotStarted` but not process `Starting`
So `Starting` could confuse

Reply:
Ok, I agree that consistency is important and a state should describe a state.

Maybe nesting is confusing me:

- `Initialized.NotStarted`
- `Initialized.Started` - two synonyms in one state

Do clients use this state `Initialized` by itself? Didn't find.

Reply:
If not, I'd propose to merge them into two: `Initialized` and `Started`.